### ISSUE TITLE
Navbar not styling <ul> and <li>

### ISSUE THREAD

**Issue by ajcerejeira** [jgthms/bulma#1307]
### Overview of the problem

This is about the Bulma **CSS framework**

My **browser** is: `Firefox 56.0 (64-bit) Ubuntu`

### Description

When I try to use navbar with `<ul>` and `<li>` the `nav-items` do not get styled

I have the following `HTML` code:

```html
<nav class="navbar is-primary">
  <h1 class="navbar-brand"><a href="/" class="navbar-item">Children Of U</a></h1>

  <ul class="navbar-menu navbar-end is-active">    
    <li ><a class="navbar-item" href="/blog.html">Blog</a></li>
    <li ><a class="navbar-item" href="/team.html">Voluntários</a></li>
    <li ><a class="navbar-item" href="/gallery.html">Galeria</a></li>
    <li class="active"><a class="navbar-item" href="/contact.html">Contacto</a></li>
  </ul>
</nav>
```

### Expected behavior

If i simply remove the `<li>` tags it works:

```html
<nav class="navbar is-primary">
  <h1 class="navbar-brand"><a href="/" class="navbar-item">Children Of U</a></h1>

  <ul class="navbar-menu navbar-end is-active">    
    <a class="navbar-item" href="/blog.html">Blog</a>
    <a class="navbar-item" href="/team.html">Voluntários</a>
    <a class="navbar-item" href="/gallery.html">Galeria</a>
    <a class="navbar-item" href="/contact.html">Contacto</a>
  </ul>
</nav>
```

![captura de ecra de 2017-10-13 15-06-34](https://user-images.githubusercontent.com/22506305/31550110-23a6419c-b028-11e7-8388-ca3a7aec7e31.png)

### Actual behavior

![captura de ecra de 2017-10-13 14-58-35](https://user-images.githubusercontent.com/22506305/31550026-dfd26586-b027-11e7-80ae-149d4c8d8ebb.png)

### Offtopic

Is there any way to style `active` links (like in `Bootstrap`)?

**Comment by jgthms** [comment]
You can't do this:

```
<ul class="navbar-menu navbar-end is-active"> 
```

The `navbar-end` should be a direct child of `navbar-menu`.

```
<div class="navbar-menu">
  <ul class="navbar-end"> 
    <li class="navbar-item"><a href="/blog.html">Blog</a></li>
```

**Comment by ajcerejeira** [comment]
Thank you for the reply, 
I tried the following:

```html
<nav class="navbar is-primary">
  <h1 class="navbar-brand"><a href="/" class="navbar-item">Children Of U</a></h1>

  <div class="navbar-menu">
    <ul class="navbar-end is-active"> 
      <li class="navbar-item"><a href="/blog.html">Blog</a></li>
      <li class="navbar-item"><a href="/team.html">Voluntários</a></li>
      <li class="navbar-item"><a href="/gallery.html">Galeria</a></li>
      <li class="navbar-item"><a href="/contact.html">Contacto</a></li>
    </ul>
  </div>
</nav>
```

But now the navbar doesn't show any links, only on hover:

![captura de ecra de 2017-10-14 14-08-53](https://user-images.githubusercontent.com/22506305/31575959-3d05e07c-b0e9-11e7-8c36-9c971445721d.png)

**Comment by jgthms** [comment]
Ah yes, AFAICT the `navbar` doesn't support `li`. You need to have:

```
<div class="navbar-end">
  <a class="navbar-item" href="/blog.html">Blog</a>
</div>
```

**Comment by Log1x** [comment]
I personally use `@extend` because of this-- but it's not something I think is good practice.

I do think you should consider supporting `<nav><ul><li><a></a></li></ul></nav>` as its the structure of every nav example in existence as well as examples of Schema usage.

**Comment by LucyI** [comment]
+1 for supporting a **nav-ul-li-a** navbar structure. Necessary for using Bulma with almost any CMS or portal framework I can think of...

**Comment by njwest** [comment]
+1, Bulma's nav opinions clash with the opinions of some CMSes/front-end libraries in a challenging way

**Comment by caiquecastro** [comment]
+1 for this.

**Comment by UisgeBeatha436** [comment]
+1 Is there any update?

**Comment by sjclark** [comment]
The `ul > li > a` structure is really the only correct ARIA approach to this, slightly frustrating it doesn't work correctly.

**Comment by patie** [comment]
any news?

**Comment by RobertSK** [comment]
any news?
this is important!

**Comment by Sarathlal** [comment]
Any updates?

**Comment by ajcerejeira** [comment]
I know this issue as been automatically marked as `closed`, however I still think that the `ul > li > a` structured should be properly supported since it looks more semantic (we are effectively building a "list" of "links").

Do we now what is currently preventing `bulma` to support this?

**Comment by sjclark** [comment]
@Log1x could you maybe post the code you're using to *extend* this?

**Comment by MarkoSkofljanec** [comment]
I added following SCSS to my Vue component which uses a navbar:

/* Q&D fix for BULMA 0.8 add style for vue router-link with tag="li" */
li.navbar-item {
  @extend .navbar-item;
  cursor: pointer;
}

/* FIXME: add customize variables */
li.navbar-item:hover {
  @extend .navbar-item;
  background-color: #f2f2f2;
  color: #0a0a0a;
}

**Comment by PDXMan1** [comment]
This appears to be working. 

`<div class="navbar-menu">
	<ul class="navbar-end">

		<li class="navbar-item">
			<a>Home</a>
		</li>

		<li class="navbar-item">
			<a>Channel Tab</a>
		</li>

		<li class="navbar-item has-dropdown is-hoverable">
			<a class="navbar-link">Second Channel Tab</a>

				<ul class="navbar-dropdown">
					<li class="navbar-item"><a>About</a></li>
					<li class="navbar-item"><a>Jobs</a></li>
					<li class="navbar-item"><a>Contact</a></li>
				</ul>
				
		</li>

	</ul>{!-- /.navbar-end --}
</div>{!-- /.navbar-menu --}`
